README.md

```diff
@@ -1 +1,141 @@
+
+#### English
+All code, names, comments, etc. must be in English.
+> Tip: install a spell checker in your IDE to avoid typos.
+
+#### Use one name for one thing
+Do not reuse names. Do not use names that can mean multiple things. Always use the same name for the same thing. So use the same name within the JavaScript, CSS and HTML for the same thing. Align naming with the whole team: Backend, Frontend, UX, Design, PM, client, etc.
+
+For big projects with their own jargon it could help to create a dictionary.
+
+#### Descriptive
+Names must be descriptive for the working or usage of the class, method or variable.
+
+###### The name should always end with what the object is.
+For example: a button must always end with `Button`.
+ * `MenuButton` a button used in the menu
+ * `ButtonMenu` a menu with buttons
+ * `MenuButtonIcon` an icon in the button of the menu
+ 
+There is no limit for the length of a name, so prefer a long name which is clear and descriptive than a short name which is not clear.
+
+#### Abbreviations
+Avoid them as a general rule. For example, `calculateOptimalValue()` is a better method name than `calcOptVal()`. Being clear is more important than minimizing keystrokes. If you don't abbreviate, developers won't have to remember whether you shortened a word like _qualified_ to `qual` or `qlfd`.
+
+Abbreviations that are part of the HTML, CSS and/or JavaScript API are allowed, like:
+ * `URL` for Uniform Resource Locator
+ * `URI` for Uniform Resource Identifier
+ * `src` for source
+ * `DOM` for Document Object Model
+ * `img` for image
+ 
+You should only use these abbreviations within the same context. So only use `img` if you refer to an HTML Image Tag. (`<img>`)
+
+We have standardized on a few abbreviations that are allowed to use:
+ * `app` for application
+ * `args` for arguments
+ * `auto` for automatic, as in `autoLayout`
+ * `bin` for binary
+ * `fps` for frames per second
+ * `id` for identifier. Please note that 'd' should be written in lowercase when used in combination with an other word, like `userId`.
+ * `info` for information, as in `GridRowInfo`
+ * `init` for initialize
+ * `lib` for library
+ * `max` for maximum, as in `maxHeight`
+ * `min` for minimum, as in `minWidth`
+ * `param` for parameter
+ * `params` for parameters
+ * `ref` for reference
+ * `regExp` for regular expression, as in `RegExpValidator`
+ * `temp` for temporary
+ * `ui` for user interface
+ * `util` or `utils` for utility or utilities, as in `StringUtils`
+
+#### Plural or singular?
+
+##### Classes
+When you would instantiate or type the class it must have singular name, also for enums.
+When you should not instantiate the class, it should have a plural name.
+
+##### Arrays
+or other kind of lists should have a plural name or end with `List`, like `userList`.
```

Comment:
I like more the word "Collection", for me the word List is a bit miselading because in some languages an array and a list are different things. I know this is Javascript but I think it doesn't hurt that much to use the word "Collection"

Reply:
I like the word `Collection` too. So we could add that option too.

However this could also be misleading because of the same reason. "Collection" means something in some languages. :-)

Reply:
MDN talks about [indexed](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Indexed_collections) and [keyed](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Keyed_collections) **Collections** (array, object, map, set), so I don't think it would be that confusing.

Reply:
Although, in most cases it could be beneficial to be a bit more specific, for example:

* `userMap` implies that it's a keyed map, possible by `id`, to make lookup easier.
* `userList` implies that it is a list, and there is some order to it, and you can directly iterate over the values.
* `users` is also just fine in certain context where there is no ambiguity.